zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/user/UserDeleteProcessor.java

```diff
@@ -148,6 +148,8 @@ private void deleteUser(final PersistedUser user) {
     }
 
     stateWriter.appendFollowUpEvent(
-        userKey, UserIntent.DELETED, new UserRecord().setUserKey(userKey));
+        userKey,
+        UserIntent.DELETED,
+        new UserRecord().setUserKey(userKey).setUsername(user.getUsername()));
```

Comment:
🔧 It might make more sense to just return the entire record. It's not much data so I'm not worried about running out of the maximum message size. So it'd be `user.getUser()`.

Reply:
Lets do that, I'll update it now